### PR TITLE
Concord config

### PR DESCRIPTION
**Pull request by RBlafford**
- New program for cli that contains 3 subprograms: init, set, show
- All exiting concord cli programs will look for defaults in the first .concord.cfg file that exists in cd or any directory above it

### CODE REVIEW

File: concord_cli/config.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import os
+import json
+import argparse
+from functools import partial
+
+CONFIG_FILENAME = '.concord.cfg'
+
+CONCORD_DEFAULTS = { 'zookeeper_path' : '/bolt',
```
Comment (RBlafford) [inline concord_cli/config.py]: probably should change all /bolt to /concord

File: concord_cli/tracer.py [hunk 2]
```diff
@@ -32,6 +32,10 @@ def generate_options():
     return parser
 
 def validate_options(options, parser):
+    config = default_options()
+    if config is not None:
+        options.zookeeper = config['zookeeper_hosts']
+        options.zk_path = config['zookeeper_path']
```
Comment (RBlafford) [inline concord_cli/tracer.py]: This code is repeated many times. Move into default_options and have it done once. Ye.

File: concord_cli/config.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+def find_config(src):
+    """ recursively searches .. until it finds a file named CONFIG_FILENAME
+    will return None in the case of no matches or the abspath if found"""
+    if src == '':
+        return None
+    test_file = os.path.join(src, CONFIG_FILENAME)
+    if os.path.isfile(test_file):
+        return test_file
+    return find_config(src[0:src.rfind('/')])
```
Comment (RBlafford) [inline concord_cli/config.py]: use os.path.dirname()

File: concord_cli/tracer.py [hunk 4]
```diff
@@ -19,12 +19,12 @@
...
                       action="store", help="i.e: 1.2.3.4:2181,2.3.4.5:2181")
     parser.add_option("--file", help="output file",
                       default="trace_graph", action="store", dest="filename")
-    parser.add_option("--scheduler-path", help="zookeeper path, i.e.: /bolt",
-                      default="/bolt", action="store", dest="zk_path")
+    parser.add_option("--scheduler-path", help="zookeeper path, i.e.: /concord",
+                      default="/concord", action="store", dest="zk_path")
     parser.add_option("--trace-id", help="any trace id, i.e.: 1121345",
                       action="store", dest="trace_id")
     parser.add_option("--scheduler-address", help="i.e.: 1.2.3.4:11219",
```
Comment (senior7515) [inline concord_cli/tracer.py]: fix

**Pull request merged**